### PR TITLE
Repair wrong path to Poll, mods in close method

### PR DESCRIPTION
**Pull request by jankowskirobert**
I have issue like a below:
Can't locate Net/OpenSSH/Compat/SSH/Poll.pm in @INC (you may need to install the Net::OpenSSH::Compat::SSH::Poll
...
at /opt/perl-5.18.3/lib/site_perl/5.18.2/Net/OpenSSH/Compat/SSH2.pm line 305.

also i got uninitialized error while trying to use close method in SSH2.pm line 476

### CODE REVIEW

**Comment by jankowskirobert** [comment]
Any reaction ?

File: lib/Net/OpenSSH/Compat/SSH2.pm [hunk 1]
```diff
@@ -466,14 +466,14 @@ sub _slave_exited {
 sub DESTROY {
     my $chan = shift;
     my $ch = $chan->_hash;
-    $chan->close if $ch->{state} eq 'exec';
+    $chan->close if $ch->{state} eq 'exec' and defined $chan;
```
Comment (salva) [inline lib/Net/OpenSSH/Compat/SSH2.pm:469]: why the `and defined $chan`?

File: lib/Net/OpenSSH/Compat/SSH2.pm [hunk 2]
```diff
@@ -466,14 +466,14 @@ sub _slave_exited {
...
     $chan->SUPER::DESTROY;
 }
 
 sub wait_closed {
     my $chan = shift;
     my $ch = $chan->_hash;
-    shift->close if $ch->{state} eq 'exec';
+    shift->close if $ch->{state} eq 'exec' and defined shift;
```
Comment (salva) [inline lib/Net/OpenSSH/Compat/SSH2.pm:476]: and why the "and defined shift"?

**Comment by salva** [comment]
Sorry for not replying to this before, I was on holidays for most of August and haven't caught up yet with my CPAN mails.

I have made a couple of comments on the patch: https://github.com/salva/p5-Net-OpenSSH-Compat/pull/2

File: lib/Net/OpenSSH/Compat/SSH2.pm (hunk 2 above)
Comment (jankowskirobert) [reply lib/Net/OpenSSH/Compat/SSH2.pm:476]: i got message which says: "Can't call method "close" on an undefined value at /usr/local/lib/site_perl/5.24.0/Net/OpenSSH/Compat/SSH2.pm line 476." in this lines.

File: lib/Net/OpenSSH/Compat/SSH2.pm (hunk 1 above)
Comment (jankowskirobert) [reply lib/Net/OpenSSH/Compat/SSH2.pm:469]: i got message which says: "Can't call method "close" on an undefined value at /usr/local/lib/site_perl/5.24.0/Net/OpenSSH/Compat/SSH2.pm line 476." in this lines.

File: lib/Net/OpenSSH/Compat/SSH2.pm (hunk 2 above)
Comment (salva) [reply lib/Net/OpenSSH/Compat/SSH2.pm:476]: ah, ok, but the problem is actually using `shift` again on the left of the `close` method invokation.

**Comment by salva** [comment]
Can you try the version at master now?

I have (hopefully) fixed both problems there.

**Comment by jankowskirobert** [comment]
yeap, its working now, Thanks!

**Pull request closed without merging**